draft-ietf-httpbis-rfc6265bis.md

```diff
@@ -854,9 +854,20 @@ cookie        = cookie-string
 cookie-string = cookie-pair *( ";" SP cookie-pair )
 ~~~
 
+While {{Section 5.4 of RFC9110}} does not define a length limit for header
+fields it is likely that the web server's implementation does impose a limit;
+many popular implementations have default limits of 8K. Server SHOULD avoid
+setting a large number of large cookies such that the final cookie-string
+would exceed their header field limit. Not doing so could result in requests
+to the server failing.
+
 Servers MUST be tolerant of multiple cookie headers. For example, an HTTP/2
 {{RFC9113}} or HTTP/3 {{RFC9114}} connection might split a cookie header to
-improve compression.
+improve compression. Servers are free to determine what form this tolerance
+takes. For example, the server could process each cookie header individually
+or the server could concatenate all the cookie headers into one and then
+process that final, single, header. The server should be mindful of any
+header field limits when deciding which approach to take.
```

Comment:
It might be besides the point right at this point in the text, but something common in my experience is for a reverse proxy to do H2 or H3 termination. It might tend to pick some strategy such as merging, in order to build a compliant single cookie that is forwards. 

Such a proxy might not validate the cookie in any way - not length nor content. Instead, it passes on this jumbo cookie that then fails at the origin.

This can be surprising to the origin admin. They might have followed the set-cookie advise and might control the UA to ensure that cookie splitting is happening. But somewhere along the chain the cookies are getting merged, leading to some size limit check somewhere failing.

Reply:
If we want to address it here, perhaps something like:

> For example, the server could process each cookie header individually. Alternatively, the server or an intermediary could concatenate all the cookie headers into one and then process/forward that final, single, header. The server and any intermediaries should be mindful of any header field limits when deciding which approach to take.

Reply:
I added a note that intermediaries should make large headers. While it'd be great if intermediary implementation followed that advice I think there is a benefit in giving a server's admin a hint as to why their requests might be failing and where to trying troubleshooting.